File: business-platform-web/src/types/service.ts
export interface ServiceListDto {
  id: number;
  name: string | null;
  slug: string | null;
  shortDescription: string | null;
  primaryImageUrl: string | null;
}

export interface ServiceImageDto {
  id: number;
  imageUrl: string | null;
  altText: string | null;
  isPrimary: boolean;
  displayOrder: number;
}

export interface ServiceDetailsDto {
  id: number;
  name: string | null;
  slug: string | null;
  shortDescription: string | null;
  description: string | null;
  images: ServiceImageDto[] | null;
}

File: business-platform-api/Models/Entities/Service.cs
namespace business_platform_api.Models.Entities;

public class Service : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string? ShortDescription { get; set; }
    public string? Description { get; set; }
    public bool IsActive { get; set; } = true;
    public int DisplayOrder { get; set; } = 0;

    public ICollection<ServiceImage> Images { get; set; } = new List<ServiceImage>();
}
